### PR TITLE
Solve Problems with jQuery [Merge this first]

### LINKED ISSUE #66: send2cgeo on old map stops opening the cache details
When the script is active then the click on a cache icon has no result.
Without the script it shows the cache info.

### LINKED ISSUE #81: GCLH2 does not work while send2cgeo got new update
Today i got the update from sent2cgeo and now my GCLH2 does not work any longer.
If i stop the send2cgeo-Script everything is ok.
Please check

Win10, Firefox 72.0.2

### LINKED ISSUE #82: Non-functional change of coordinates
You cannot change coordinates when the script is on. nothing happens when you click Accept. Tested on multiple geocaches. Everything is OK after turning off the script.

### PR DESCRIPTION
**Pull request by capoaira**
Fix #66 
Fix #81 
Fix #82 

If you test the browse (old) map, the language of geocaching.com have to be english. Because #68 is not fixed now.

### CODE REVIEW

**Comment by capoaira** [comment]
Maybe we can do a new release, if this PR is merged...

File: send2cgeo.user.js [hunk 1]
```diff
@@ -25,27 +24,163 @@
...
         $("#send2cgeo, #send2cgeo div").fadeIn();
         // hide iframe for now and wait for page to be loaded
         $("#send2cgeo iframe")
             .hide()
             .off('load')
-            .attr('src', 'https://send2.cgeo.org/add.html?cache=' + code)
+            .attr('src', 'https://send2.cgeo.org/add.html?cache=' + GCCode)
             .on('load',
                 function() {
                     // hide "please wait text" and show iframe
```
Comment (bekuno) [inline send2cgeo.user.js]: Why you doubled the code (here and in window.s2geo = function(code) {}?

File: send2cgeo.user.js [hunk 2]
```diff
@@ -25,27 +24,163 @@
...
+    // function that handles the actual sending
+    $('#s2cg-' + GCCode).bind('click', function(){
         // show the box and the "please wait" text
         $("#send2cgeo, #send2cgeo div").fadeIn();
         // hide iframe for now and wait for page to be loaded
         $("#send2cgeo iframe")
             .hide()
             .off('load')
-            .attr('src', 'https://send2.cgeo.org/add.html?cache=' + code)
+            .attr('src', 'https://send2.cgeo.org/add.html?cache=' + GCCode)
```
Comment (bekuno) [inline send2cgeo.user.js]: Please refaktor: only use "code" or "GCCode" in the script for the same meaning.

File: send2cgeo.user.js (hunk 1 above)
Comment (capoaira) [reply send2cgeo.user.js]: If I use the the window.s2geo() function it does not work on the searchmap. If I use the function addSend2cgeoButton() it does not work at the Browsemap and new seach.
But I think that I could shorten the code if I don't make it as a function, rather integrate it directly into the search map.

**Comment by capoaira** [comment]
Sometimes I does't understand. When I try it first it works, now it doesn't work. I have to look again...

> Two points from ESLint:
> 
>     * Line 113 - 'quitOnAdFrames' was used before it was defined.
> 
>     * Line 123: 'c' is defined but never used.
> 
> 
> ```
> var quitOnAdFrames = function(c) {
> ```

I made the changes

**Comment by capoaira** [comment]
I've tested and it works now.

**Comment by capoaira** [comment]
@bekuno I've check all features and everythink work for me. Please have a second look if you have time.
@Lineflyer When this PR is merged, all functions work again. Can you then merge the master branch into release?

**Comment by mucek4** [comment]
Codacy says there shall be some double quotes.

**Comment by capoaira** [comment]
> Codacy says there shall be some double quotes.

I can change that, but it doesn't make sense to me. I used single quotes throughout the script. Codacy wants to change it to double quotes in just four lines.

**Comment by bekuno** [comment]
It is rebased and squashed now.
I will test it the next days deeper.

File: send2cgeo.user.js [hunk 3]
```diff
@@ -25,107 +24,130 @@
...
+    + '    );'
+    + '}'
+
+    + 'function s2geomultiProcessLine(toAddIframe, skipFound, requestedCnt, alreadySent) {'
+    + '    if (skipFound) {'
+    + '        var GCCode = toAddIframe.attr("send2cgeo_gccode");'
+    + '        toAddIframe.html("<iframe width=120 height=80 src=\'https://send2.cgeo.org/add.html?cache=" + GCCode + "\'>");'
+    + '        toAddIframe.attr("send2cgeo_sent", "1");'
+    + '        s2cgeoProgressReport((alreadySent + 1) + " ", alreadySent !== 0);'
+    + '    }'
```
Comment (bekuno) [inline send2cgeo.user.js]: original:
~~~
  function s2geomultiProcessLine(toAddIframe, skipFound, requestedCnt, alreadySent) {
      if (skipFound) {
          if (toAddIframe.parent().parent().find("use[xlink\\:href*=#icon-found]").length === 0) {
              var code = toAddIframe.attr("send2cgeo_gccode");
              toAddIframe.html("<iframe width=120 height=80 src=\"https://send2.cgeo.org/add.html?cache="+code+"\">");
              toAddIframe.attr("send2cgeo_sent","1");
              s2cgeoProgressReport((alreadySent+1)+" ",alreadySent!==0);
          } else {
              var trToDel = toAddIframe.parent().parent();
              trToDel.remove();
              s2cgeoProgressReport(".",alreadySent!==0);
          }
      }
~~~
Which problem was solved by skipping some lines here?

File: send2cgeo.user.js [hunk 4]
```diff
@@ -25,107 +24,130 @@
...
+    + '        100'
+    + '    );'
+    + '};';
+
+document.getElementsByTagName('head')[0].appendChild(s2geomultiScript);
+
+// This solves the problems with jquery
+var quitOnAdFrames = function() {
+    var quitOnAdFramesDeref = new jQuery.Deferred();
+    if(window.name) {
```
Comment (bekuno) [inline send2cgeo.user.js]: add space:
~~~
    if (window.name) {
~~~

File: send2cgeo.user.js [hunk 5]
```diff
@@ -25,107 +24,130 @@
...
+        if (window.name.substring(0, 18) !== 'google_ads_iframe_') {
+            quitOnAdFramesDeref.resolve();
+        }else {
+            quitOnAdFramesDeref.reject();
         }
+    }else {
```
Comment (bekuno) [inline send2cgeo.user.js]: add space:
~~~
    } else {
~~~

File: send2cgeo.user.js [hunk 6]
```diff
@@ -25,107 +24,130 @@
...
+
+document.getElementsByTagName('head')[0].appendChild(s2geomultiScript);
+
+// This solves the problems with jquery
+var quitOnAdFrames = function() {
+    var quitOnAdFramesDeref = new jQuery.Deferred();
+    if(window.name) {
+        if (window.name.substring(0, 18) !== 'google_ads_iframe_') {
+            quitOnAdFramesDeref.resolve();
+        }else {
```
Comment (bekuno) [inline send2cgeo.user.js]: add space:
~~~
      } else {
~~~

File: send2cgeo.user.js [hunk 7]
```diff
@@ -276,15 +298,34 @@ s.textContent = '(' + function() {
...
+                            $('.s2cg').remove();
                         }
-                        // Add c2cg button.
-                        var html = '<li class="c2cg"><a style="cursor:pointer;" id="s2cg-' + GCCode
-                            + '" onclick="window.s2geo(\'' + GCCode + '\'); return false;">'
-                            + '<img class="action-icon" src="https://send2.cgeo.org/send2cgeo.png" />'
-                            + '<span>Send to c:geo</span></a></li>';
+                        // Add s2cg button.
+                        var html = '<li class="s2cg"><a id="s2cg-' + GCCode + '" href="javascript:void(0);" title="Send to c:geo">'
+                            + '    <img class="action-icon" src="https://send2.cgeo.org/send2cgeo.png">'
```
Comment (bekuno) [inline send2cgeo.user.js]: please close the ""

File: send2cgeo.user.js [hunk 8]
```diff
@@ -276,15 +298,34 @@ s.textContent = '(' + function() {
...
+                            + '    <span>Send to c:geo</span>'
```
Comment (bekuno) [inline send2cgeo.user.js]: There are no spaces in front of a line on the other html vars.

File: send2cgeo.user.js [hunk 9]
```diff
@@ -276,15 +298,34 @@ s.textContent = '(' + function() {
...
+                            + '</a></li>'
                         $('.more-info-li').before(html);
+
+                        // On the new searchmap the window.s2geo() doesn't work.
```
Comment (bekuno) [inline send2cgeo.user.js]: Please mark this code copy here as a workaround more clearly.

**Comment by bekuno** [comment]
Please describe the full changes in your next commit. (Headline and text)
Then it is easier to have a usable commit text later in the git history.

File: send2cgeo.user.js (hunk 3 above)
Comment (capoaira) [reply send2cgeo.user.js]: Maybe it was lost during copy and paste ... Sorry

**Comment by capoaira** [comment]
I added two commits that will (hopefully) resolve your requested changes.

**Comment by bekuno** [comment]
@capoaira Many thanks for this big work.

**Pull request merged**